.github/workflows/release.yml

```diff
@@ -42,23 +48,133 @@ jobs:
           ./hack/build-and-publish-release.sh
           mkdir release
           cp ./tmp/release.yml release/
+      - name: Sign secretgen-controller OCI image
+        run: |
+          image_url=`yq e '.spec.template.spec.containers[] | select(.name == "secretgen-controller") | .image' release/release.yml`
+          cosign sign --yes "$image_url"
+      - name: Verify signature on secretgen-controller OCI image
+        run: |
+          image_url=`yq e '.spec.template.spec.containers[] | select(.name == "secretgen-controller") | .image' release/release.yml`
+          cosign verify \
+            $image_url \
+            --certificate-identity-regexp=https://github.com/${{ github.repository_owner }} \
+            --certificate-oidc-issuer=https://token.actions.githubusercontent.com
       - name: Run Package build
         run: |
           constraintVersion="${{ github.ref_name }}"
           kctrl pkg release -y -v ${constraintVersion:1} --debug
           mv carvel-artifacts/packages/secretgen-controller.carvel.dev/metadata.yml carvel-artifacts/packages/secretgen-controller.carvel.dev/package-metadata.yml
           cp carvel-artifacts/packages/secretgen-controller.carvel.dev/* release/
-      - name: Checksum
+      - name: Sign secretgen-controller-package-bundle OCI image
+        run: |
+          image_url=`yq e '.spec.template.spec.fetch[0].imgpkgBundle.image' release/package.yml`
+          cosign sign --yes "$image_url"
+      
+      - name: Verify signature on secretgen-controller-package-bundle OCI image
+        run: |
+          image_url=`yq e '.spec.template.spec.fetch[0].imgpkgBundle.image' release/package.yml`
+          cosign verify \
+            $image_url \
+            --certificate-identity-regexp=https://github.com/${{ github.repository_owner }} \
+            --certificate-oidc-issuer=https://token.actions.githubusercontent.com
+
+      - name: Updating release notes
+        run: |
+          RELEASE_TAG=$(git describe --tags --abbrev=0)
+          KAPP_CONTROLLER_IMAGE=$(yq e '.spec.template.spec.containers[] | select(.name == "secretgen-controller") | .image' release/release.yml)
```

Comment:
The variables referred in this job are still referring to Kapp controller.

Reply:
fixed